### PR TITLE
Add easy upgrades support

### PR DESCRIPTION
**Pull request by spiccinini**

### CODE REVIEW

**Comment by ilario** [comment]
Sounds good, can you tell us what is this?
Is this related to #325?

**Comment by spiccinini** [comment]
> Sounds good, can you tell us what is this?
> Is this related to #325?

This is related, yes. This package provides ubus and commands for downloading and validating signatures of upgrades. The mechanism is somehow similar to unnatended upgrades but without the ability to request specific builds dinamically.

File: packages/eupgrade/files/usr/lib/lua/eupgrade.lua [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+eup.STATUS_DEFAULT = 'not-initiated'
+eup.STATUS_DOWNLOADING = 'downloading'
+eup.STATUS_DOWNLOADED = 'downloaded'
+eup.STATUS_DOWNLOAD_FAILED = 'download-failed'
+
+
+local uci = libuci.cursor()
+
+function eup.set_workdir(workdir)
+    if not utils.file_exists(workdir) then
```
Comment (G10h4ck) [inline packages/eupgrade/files/usr/lib/lua/eupgrade.lua:16]: what if exists but it is not dir?

File: packages/eupgrade/Makefile [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+include ../../libremesh.mk
+
+PKG_NAME:=eupgrade
+
+define Package/$(PKG_NAME)
+  SECTION:=utils
+  CATEGORY:=Utilities
+  TITLE:=$(PKG_NAME) provides semi automated firmware upgrades
+  MAINTAINER:=Santiago Piccinini <spiccinini@altermundi.net>
+  DEPENDS:=+lua +lime-system
```
Comment (G10h4ck) [inline packages/eupgrade/Makefile:16]: Is this yet another package which only on `lime.utils`?

**Review by G10h4ck — APPROVED** [review]

File: packages/eupgrade/Makefile (hunk 2 above)
Comment (spiccinini) [reply packages/eupgrade/Makefile:16]: Yes!

File: packages/eupgrade/files/usr/lib/lua/eupgrade.lua (hunk 1 above)
Comment (spiccinini) [reply packages/eupgrade/files/usr/lib/lua/eupgrade.lua:16]: What do you suggest, to loga warning if it is not a directory?

Comment (G10h4ck) [reply packages/eupgrade/files/usr/lib/lua/eupgrade.lua:16]: in that case it probably should fail all together and report an error

**Review by germanferrero — APPROVED** [review]

**Pull request merged**